### PR TITLE
[14.0][FIX] s_env_data_encryption: fix access right issues when using actions

### PR DESCRIPTION
**Pull request by sebalix**

### CODE REVIEW

File: server_environment_data_encryption/models/server_env_mixin.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def action_change_env_data_encrypted_fields(self):
             # We don't know which action we are using... take default one
             action = self.get_formview_action()
         else:
-            action = self.env["ir.actions.act_window"].browse(action_id).read()[0]
+            action = self.env["ir.actions.act_window"]._for_xml_id(action_id)
```
Comment (florian-dacosta) [inline server_environment_data_encryption/models/server_env_mixin.py]: @sebalix Not sure about this one.
`action_id` is the database id while `_for_xml_id` takes an `xml_id` as an input.

I guess using sudo in the read would be better

Comment (sebalix) [reply server_environment_data_encryption/models/server_env_mixin.py]: Oups you are right, I was a bit fast...

Comment (sebalix) [reply server_environment_data_encryption/models/server_env_mixin.py]: Fixed

**Review by florian-dacosta — APPROVED** [review]

**Comment by florian-dacosta** [comment]
@sebalix Ok now thanks
Just a pre-commit to run to fix the bot issue

**Comment by sebalix** [comment]
pre-commit fixed but I don't know why Runboat is failing here

**Review by i-vyshnevska — APPROVED** [review]

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
/ocabot merge patch

The error seems unrelated. @florian-dacosta can you check?

```
2022-02-02 06:42:16,433 266 CRITICAL b522ebfa5-b291-4fd6-8330-44cd0353ae52 odoo.service.server: Failed to initialize database `b522ebfa5-b291-4fd6-8330-44cd0353ae52`. 
Traceback (most recent call last):
  File "/opt/odoo/odoo/tools/convert.py", line 677, in _tag_root
    f(rec)
  File "/opt/odoo/odoo/tools/convert.py", line 580, in _tag_record
    record = model._load_records([data], self.mode == 'update')
  File "/opt/odoo/odoo/models.py", line 4217, in _load_records
    data['record']._load_records_write(data['values'])
  File "/opt/odoo/odoo/models.py", line 4146, in _load_records_write
    self.write(values)
  File "/opt/odoo/addons/payment_transfer/models/payment.py", line 68, in write
    return super(TransferPaymentAcquirer, self).write(values)
  File "/opt/odoo/addons/payment/models/payment_acquirer.py", line 335, in write
    result = super(PaymentAcquirer, self).write(vals)
  File "/mnt/data/odoo-addons-dir/setup/server_environment/odoo/addons/server_environment/models/server_env_tech_name_mixin.py", line 66, in write
    return super().write(vals)
  File "/opt/odoo/odoo/models.py", line 3704, in write
    fields[0].determine_inverse(real_recs)
  File "/opt/odoo/odoo/fields.py", line 1187, in determine_inverse
    getattr(records, self.inverse)()
  File "/mnt/data/odoo-addons-dir/setup/server_environment_data_encryption/odoo/addons/server_environment_data_encryption/models/server_env_mixin.py", line 50, in _inverse_server_env
    encrypted_data_name, values, env=env
  File "/mnt/data/odoo-addons-dir/setup/data_encryption/odoo/addons/data_encryption/models/encrypted_data.py", line 145, in _encrypted_store_json
    return self._encrypted_store(name, json.dumps(json_data), env=env)
  File "/mnt/data/odoo-addons-dir/setup/data_encryption/odoo/addons/data_encryption/models/encrypted_data.py", line 132, in _encrypted_store
    encrypted_data = self._encrypt_data(data, env)
  File "/mnt/data/odoo-addons-dir/setup/data_encryption/odoo/addons/data_encryption/models/encrypted_data.py", line 119, in _encrypt_data
    cipher = self._get_cipher(env)
  File "/mnt/data/odoo-addons-dir/setup/data_encryption/odoo/addons/data_encryption/models/encrypted_data.py", line 111, in _get_cipher
    % (key_name, Fernet.generate_key())
odoo.exceptions.ValidationError: No 'encryption_key_test' entry found in config file. Use a key similar to: b'yD1RtwcBHAkVWoe4vACu4qJ2AQvRHJGM-aCoQBUGkXs='
```

**Pull request merged**